src/system/SystemLayer.cpp

```diff
@@ -31,5 +31,14 @@ CHIP_ERROR Layer::ScheduleLambdaBridge(LambdaBridge && bridge)
     return lReturn;
 }
 
+CHIP_ERROR Layer::RunOnMatterContext(std::function<CHIP_ERROR()> func)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    PlatformEventing::LockMatterStack(*this);
```

Comment:
LockMatterStack as implemented in this PR can be a no-op on some platforms, so cannot be used for anything that actually happens cross-platform.

Reply:
I have moved it to LayerFreeRTOS.